train.py

```diff
@@ -1,216 +1,228 @@
-                                        # 1 = accuracy training 
-                                        # 2 = extended training
-    batch_size = args.batch_size 
-    test_size = 0.2
-    epochs = args.max_epochs
-    input_size = args.input_size
-
-    # TRAINING MODE
-    if train_mode == 0:         # Speed Training
-        learning_rate = 0.01 
-        conv1d_strides = 12    
-        conv1d_filters = 16
-        hidden_units = 36
-    elif train_mode == 1:       # Accuracy Training (~10x longer than Speed Training)
-        learning_rate = 0.01 
-        conv1d_strides = 4
-        conv1d_filters = 36
-        hidden_units= 64
-    else:                       # Extended Training (~60x longer than Accuracy Training)
-        learning_rate = 0.0005 
+    Modes:
+        Speed training (default)
+        Accuracy training
+        Extended training (set max_epochs as desired, for example 50+)
+    """
+    print(f"Training started")
+
+    timestamp = datetime.datetime.now().strftime("%Y-%b-%d-%H%M%S").lower()
+    name, in_files, out_files = IO.validate_source(source_dir)
+    out_dir = f'{name}_{timestamp}'
+    os.makedirs(f'models/{out_dir}')
+    test_size = test_pct/100
+
+    if training_mode == 'extended':
+         # Extended Training (~60x longer than Accuracy Training)
+        learning_rate = 0.0005
         conv1d_strides = 3
         conv1d_filters = 36
         hidden_units= 96
-
-    # Create Sequential Model ###########################################
+    elif training_mode == 'accurate':
+        # ~10x longer than Speed Training
+        learning_rate = 0.01
+        conv1d_strides = 4
+        conv1d_filters = 36
+        hidden_units= 64
+    else:
+        learning_rate = 0.01
+        conv1d_strides = 12
+        conv1d_filters = 16
+        hidden_units = 36
+    print(f'Params set for training mode "{training_mode}"')
+
+    # =========================================================================
+    # Create Sequential Model
+    # -------------------------------------------------------------------------
+    # A Sqeuential model is a linear stack of layers. "Same" padding is used to
+    # ensure the output is the same size as the input. The input shape is the
+    # number of samples (input_size) and the number of features (1). The model
+    # is compiled with the Adam optimizer and the custom loss function "esr".
+    #
+    # The history object is used to store the results of model.fit and append
+    # metadata about the training process to the saved model file.
+    # =========================================================================
     clear_session()
+    history = None
     model = Sequential()
-    model.add(Conv1D(conv1d_filters, 12,strides=conv1d_strides, activation=None, padding='same',input_shape=(input_size,1)))
-    model.add(Conv1D(conv1d_filters, 12,strides=conv1d_strides, activation=None, padding='same'))
+    model.add(Input(shape=(input_size, 1)))
```

Comment:
There was a warning or something saying that this is the preferred way to set inputs, rather than in layer args. Works fine so sure.